File: my-app/src/modules/manage/order/models/PaginatedResponse.ts
export interface PaginatedResponse<T> {
    content: T[]; // Mảng các phần tử kiểu T
    page: number; // Số trang hiện tại
    size: number; // Kích thước của mỗi trang
    totalElements: number; // Tổng số phần tử
    totalPages: number; // Tổng số trang
    lastPage: boolean; // Xác định có phải trang cuối cùng hay không
}

File: backend/logistics-orders-service/src/main/java/com/tdtu/logistics_orders_service/dto/response/PaginatedResponse.java
package com.tdtu.logistics_orders_service.dto.response;


import lombok.*;
import lombok.experimental.FieldDefaults;

import java.util.List;

@Getter
@Setter
@Builder
@ToString
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Data
public class PaginatedResponse<T> {
	private List<T> content;
	private int page;
	private int size;
	private long totalElements;
	private int totalPages;
	private boolean lastPage;

}
